codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/UnionGenerator.kt

```diff
@@ -198,13 +201,15 @@ private fun RustWriter.renderAsVariant(
             rust("if let ${unionSymbol.name}::$variantName = &self { Ok(()) } else { Err(self) }")
         }
     } else {
+        val memberSymbol = symbolProvider.toSymbol(member)
+        val targetSymbol = symbolProvider.toSymbol(model.expectShape(member.target))
         rust(
             "/// Tries to convert the enum instance into [`$variantName`](#T::$variantName), extracting the inner #D.",
             unionSymbol,
-            memberSymbol,
+            targetSymbol,
         )
         rust("/// Returns `Err(&Self)` if it can't be converted.")
-        rustBlock("pub fn as_$funcNamePart(&self) -> std::result::Result<&#T, &Self>", memberSymbol) {
+        rustBlock("pub fn as_$funcNamePart(&self) -> std::result::Result<&${memberSymbol.rustType().render()}, &Self>") {
```

Comment:
Why don't we want to render the symbol here? This looks weird.

Reply:
Member shapes are a bit weird. They represent a field on a struct/enum/union, and thus, the name is going to be that field name. Originally, I just changed this to use `targetSymbol` instead of `memberSymbol`, but that resulted in the `Box` type wrapper getting dropped on the floor, because we do boxing on the member shape instead of on the target shape... So this is actually the only correct way to do this particular action.

Reply:
You'll see a bunch of these if you code search for `rustType().render()`